### ISSUE TITLE
[AI] Infra: Docker Daemon (1 job)

### ISSUE THREAD

**Issue by dhonnappa-amd** [dhonnappa-amd/build-dashboards-testing#3287]
## AI Analysis — Infrastructure Issue: Docker Daemon

**1 affected job**

| Pipeline/Build | Queue | Job Link |
|---------------|-------|----------|
| amd-ci/7176 | `amd_mi355_1` | [View job](https://buildkite.com/vllm/amd-ci/builds/7176#019d479b-ba02-466e-be00-031eb66b274f) |

---

### Summary

This job was executing vLLM integration tests for the OpenAI API completion endpoints on AMD ROCm hardware. The test suite successfully completed the checkout phase, pulled the Docker image (rocm/vllm-ci:38606d8e6dfd42d81a339d79eafb3b7008e61c04), launched a containerized test environment, and began running pytest tests. Multiple tests passed successfully, including test_single_completion and test_no_logprobs, demonstrating that the code changes (fixing ZeroDivisionError in FP8 paged MQA logits) were working correctly. However, at 06:28:51Z, while test_zero_logprobs was in progress, the Docker daemon abruptly became unreachable with 'unexpected EOF' errors. All subsequent attempts to communicate with the daemon at tcp://127.0.0.1:2375 failed with connection errors. The container process terminated with exit code 125, which is Docker's standard error for daemon connectivity issues. This is a clear infrastructure failure—the Docker daemon either crashed, was killed by an external process, or lost network connectivity. This is NOT a test failure or code regression; the tests that ran completed successfully.

---

### Root Cause Analysis

Walking through the log chronologically: (1) The job started normally at 06:21:57Z with successful git checkout and Docker image pull. (2) The pre-command hook confirmed Docker was running ('Docker is up. Continuing...') at 06:22:41Z. (3) The test container launched successfully and began executing pytest at 06:27:05Z. (4) Tests ran successfully for approximately 1 minute and 46 seconds, with passing results logged. (5) At 06:28:51Z, the FIRST UNRECOVERABLE error occurred: 'error waiting for container: unexpected EOF' followed immediately by 'Cannot connect to the Docker daemon at tcp://127.0.0.1:2375'. This is the root cause—the Docker daemon became unreachable while the container was running. The exit code 125 confirms this was a Docker runtime failure, not a test or application error. There were no transient errors that were retried; this was a fatal infrastructure failure.

---

### Detected Events (all affected jobs)

**amd-ci/7176** (queue: `amd_mi355_1`):
- `docker_daemon`: time="2026-04-01T06:28:51Z" level=error msg="error waiting for container: unexpected EOF"
Cannot connect to the Docker daemon at tcp://127.0.0.1:2375. Is the docker daemon running?
Cannot connect to the Docker daemon at tcp://127.0.0.1:2375. Is the docker daemon running?
  - _The Docker daemon became unreachable mid-execution. The job was running tests successfully inside a Docker container when the daemon connection was lost, causing the container to fail with 'unexpected EOF' and subsequent connection attempts to fail._
- `test_failure`: entrypoints/openai/completion/test_completion.py::test_single_completion[facebook/opt-125m] PASSED
entrypoints/openai/completion/test_completion.py::test_no_logprobs[facebook/opt-125m] PASSED
  - _Tests were executing successfully before the Docker daemon failure. These passing tests demonstrate the test suite was functioning correctly._
- `docker_daemon`: The command exited with status 125
user command error: exit status 125
  - _Exit code 125 is Docker's standard error code for daemon connection failures or container runtime errors. This confirms the root cause is Docker infrastructure, not test logic._

---

#### Immediate Fix

Restart the Buildkite agent and Docker daemon on the affected node. Check system logs (journalctl -u docker) to identify why the daemon stopped. Verify the Docker socket configuration and network connectivity to tcp://127.0.0.1:2375. Retry the build job.

#### Suggested Code Change

```
No code changes required. The PR changes (Fix ZeroDivisionError in FP8 paged MQA logits for small head counts, commit 38606d8e6) are not related to this failure. Tests were passing before the Docker daemon failure.
```

#### Prevention

1. Add Docker daemon health monitoring to the Buildkite agent pre-command hook with automatic restart on failure. 2. Configure Docker with restart policies (--restart=unless-stopped). 3. Add resource limits to prevent OOM-killer from terminating the daemon. 4. Implement container-level timeouts to detect hung containers before daemon failures. 5. Add alerting for Docker daemon crashes in the CI infrastructure. 6. Consider using containerd directly instead of Docker daemon for better stability.

#### Workaround

If Docker daemon instability persists on this node, temporarily exclude it from the Buildkite agent pool using node labels/taints. Route AMD GPU jobs to a different node with a stable Docker installation. Alternatively, switch to using Kubernetes pods directly instead of Docker-in-Docker if the infrastructure supports it.

---

_This issue was automatically created by the AI log analysis pipeline._
